kernel/src/actions/mod.rs

```diff
@@ -152,6 +196,36 @@ impl Protocol {
             .as_ref()
             .is_some_and(|features| features.iter().any(|f| f == feature.as_ref()))
     }
+
+    /// Check if reading a table with this protocol is supported. That is: does the kernel support
+    /// the specified protocol reader version and all enabled reader features?
+    // TODO: should this return result so we can say _why_ it isn't supported?
+    pub fn is_read_supported(&self) -> bool {
+        if self.min_reader_version > KERNEL_READER_VERSION {
+            return false;
+        }
+        match &self.reader_features {
+            // check that all reader features are subset of supported
+            Some(reader_features) if self.min_reader_version == 3 => {
+                let parsed_features: Result<HashSet<_>, _> = reader_features
```

Comment:
Is it really an arbitrary `Result`? Or would this work?
```suggestion
                let parsed_features: DeltaResult<HashSet<_>> = reader_features
```

Reply:
we need to map from the `ReaderFeatures::from_str` Error to our DeltaError - but i'll do that explicitly so we can do what you've suggested here :)